bb_proxy/src/conn.rs

```
@@ -558,11 +599,24 @@ impl<'a, S: PacketStream + Send + Sync> Conn<'a, S> {
         }
         self.ver = ProtocolVersion::from(p.read_varint()?);
 
-        // Max len according to 1.17.1
-        let _addr = p.read_str(255)?;
-        let _port = p.read_u16()?;
-        let next = p.read_varint()?;
-        self.state = State::from_next(next);
+        match self.forwarding.as_str() {
+          "LEGACY" => {
+            // FIXME: not sure what the correct maximum length is
+            let _addr = p.read_str(2048)?;
+            let _port = p.read_u16()?;
+            let next = p.read_varint()?;
+            self.state = State::from_next(next);
+            self.info = self.read_bungeecord_info(_addr.clone());
```

In its current state, sending invalid info will just put you in offline mode. Instead, this should disallow the user from joining, so you should return an error if `read_bungeecord_info` returns `None`.

---

done